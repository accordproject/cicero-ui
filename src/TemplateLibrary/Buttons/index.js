/* React */
import React from 'react';
import PropTypes from 'prop-types';

/* Styling */
import styled from 'styled-components';
import { Button } from 'semantic-ui-react';

const UploadButton = styled.button`
  position: relative;
  font-weight: 300;
  text-align: right;
  text-decoration: none;
  font-size: 0.87em;
  color: #76777D;
  background-color: inherit;
  border: 0;
  cursor: pointer;
  &:hover {
    color: #0361DE;
    text-decoration: underline;
  }
  &:focus {
    outline: none;
    color: #3089FF;
    text-decoration: underline;
  }
  &:active {
    color: #3089FF;
    text-decoration: underline;
  }
`;

const ImportButton = styled(UploadButton)`
  margin-bottom: 5px;
`;

const AddClauseBtn = styled(Button)`
  margin: 5px 0 0 !important;
  width: 100% !important;
  max-height: 53px;
`;

export const ImportComponent = props => (
    <ImportButton onClick={props.importInput} className="importButton">
        Import from VS Code
    </ImportButton>);

export const UploadComponent = props => (
    <UploadButton onClick={props.uploadInput} className="uploadButton">
        Upload CTA file
    </UploadButton>);

export const NewClauseComponent = props => (
    <AddClauseBtn
        content="New Clause Template"
        color="blue"
        fluid
        icon="plus"
        id="addClauseBtn"
        onClick={props.addTempInput}
        className="addTemplateButton"
    />);

ImportComponent.propTypes = { importInput: PropTypes.func };
UploadComponent.propTypes = { uploadInput: PropTypes.func };
NewClauseComponent.propTypes = { addTempInput: PropTypes.func };
