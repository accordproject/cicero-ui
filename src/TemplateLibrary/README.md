## Cicero-UI - TemplateLibrary

### Usage

```shell
npm install @accordproject/cicero-ui
```

```js
import { TemplateLibrary } from '@accordproject/cicero-ui';

const libraryPropsObject = {
    ACTION_BUTTON (string),
    ACTION_BUTTON_BG (string),
    ACTION_BUTTON_BORDER (string),
    HEADER_TITLE (string),
    TEMPLATE_BACKGROUND (string),
    TEMPLATE_DESCRIPTION (string),
    TEMPLATE_TITLE (string),
}

const LibraryComponent = props => (
    <TemplateLibrary
        templates={props.templatesArray}
        upload={props.uploadCTA}
        import={props.importTemplate}
        addTemp={props.addNewTemplate}
        addToCont={props.addToContract}
        libraryProps={libraryPropsObject}
    />
);
```

### Props

- `templates` : An `array` which contains template objects with the following keys: `uir`, `name`, `version`, `description`.
- `upload` : A `function` which calls for upload functionality within the app this component is embedded in.
- `import` : A `function` which calls for import functionality within the app this component is embedded in.
- `addTemp` : A `function` which adds a new blank template to the array of templates in the Redux store of the app this component is embedded in.
- `addToCont` : A `function` which calls for adding the selected template to the Redux store in the app this component is embedded in. This will result in another component having use of its data.
- `libraryProps` : An optional `object`, see below.

### Specifications

This component is built to have the following dimensions:

```js
height: 700px;
width: 485px;
```

You can style the template card components, as well as the header. An object may be passed down this component with the following possible CSS inputs as strings:
- `ACTION_BUTTON`
- `ACTION_BUTTON_BG`
- `ACTION_BUTTON_BORDER`
- `HEADER_TITLE`
- `TEMPLATE_BACKGROUND`
- `TEMPLATE_DESCRIPTION`
- `TEMPLATE_TITLE`