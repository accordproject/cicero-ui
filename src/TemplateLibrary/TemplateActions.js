import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Icon } from 'semantic-ui-react';

const ActionsContainer = styled.div`
  padding: 0 !important;
  background-color: ${props => props.color || '#F9F9F9'} !important;
  max-height: 30px;
`;

const TemplateBtn = styled.a`
  padding: 5px 10px;
  display: inline-block;
  color: ${props => props.color || '#484848'};
  font-family: "IBM Plex Sans";
  font-size: 0.75em;
  font-weight: bold;
`;

const AddToContractBtn = styled(TemplateBtn)`
  width: 60%;
  border-right: 1px solid ${props => props.color || '#E1E5EB'}; 
  cursor: pointer;
  &:hover {
    color: #3087CB;
  }
`;

const DetailsBtn = styled(TemplateBtn)`
  float: right;
  width: 40%;
  font-size: 0.75em;
  font-weight: 300;
  text-align: center;
`;

/**
 * A Template Actions component that will provide each template
 * with functionality.
 */
class TemplateActions extends React.Component {
  /**
     * Render this React component
     * @return {*} the react component
  */
  render() {
    const { handleViewDetails, libraryProps } = this.props;
    return (
        <ActionsContainer color={libraryProps.ACTION_BUTTON_BG}>
        <div>
          <AddToContractBtn className="adToContractButton" color={libraryProps.ACTION_BUTTON_BORDER} onClick={() => this.props.addToCont(this.props.uriKey)} >
            <Icon name="plus" />
            Add to contract
          </AddToContractBtn>
          <DetailsBtn
            color={libraryProps.ACTION_BUTTON}
            onClick={() => handleViewDetails(this.props.uriKey)}>
            Details
          </DetailsBtn>
        </div>
      </ActionsContainer>
    );
  }
}

/**
 * The property types for this component
 */
TemplateActions.propTypes = {
  addToCont: PropTypes.func,
  handleViewDetails: PropTypes.func,
  uriKey: PropTypes.string,
  libraryProps: PropTypes.shape({
    ACTION_BUTTON: PropTypes.string,
    ACTION_BUTTON_BG: PropTypes.string,
    ACTION_BUTTON_BORDER: PropTypes.string,
  }),
};

export default TemplateActions;
