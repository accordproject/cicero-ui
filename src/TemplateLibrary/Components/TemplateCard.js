import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Image } from 'semantic-ui-react';

import TemplateActions from './TemplateActions';

const CardContainer = styled(Card)`
  margin: 10px 0 !important;
  border: ${props => props.tempborder || 'none'};
  border-radius: 6px !important;
  background-color: ${props => props.color || '#fff'} !important;
  text-align: left;
  box-shadow: 0 1px 9px 0 rgba(0,0,0,0.1) !important;
`;

const Title = styled.div`
  display: inline;
  color: ${props => props.color || null};
  font-size: medium;
  font-weight: 600;
  line-height: 16px;
`;

const TemplateLogo = styled(Image)`
  position: absolute !important;
  top: 13px;
  right: 16px;
  max-height: 23px;
`;

const Version = styled.span`
  margin-left: 10px;
  font-size: 0.75em;
  font-weight: 300;
`;

const DescriptionContainer = styled(Card.Description)`
  max-width: 400px;
  margin: auto;
  font-size: 0.79em;
  color: ${props => props.color || null} !important;
`;

/**
 * A Template Card component that will display the each template
 * and it's details.
 */
const TemplateCard = props => (
    <CardContainer fluid
      key={props.template.uri}
      color={props.libraryProps.TEMPLATE_BACKGROUND}
      tempborder={props.libraryProps.TEMPLATE_BORDER}
    >
        <Card.Content>
          <TemplateLogo src={props.template.icon} />
          <Title color={props.libraryProps.TEMPLATE_TITLE}>
            {
              props.template.displayName
                ? props.template.displayName
                : props.template.name
            }
            <Version>v {props.template.version}</Version>
          </Title>
          <DescriptionContainer color={props.libraryProps.TEMPLATE_DESCRIPTION}>
            {props.template.description}
          </DescriptionContainer>
        </Card.Content>
        <TemplateActions
          libraryProps={props.libraryProps}
          addToCont={props.addToCont}
          uriKey={props.template.uri}
          handleViewDetails={props.handleViewTemplate}
          className="templateAction"
        />
    </CardContainer>
);

TemplateCard.propTypes = {
  template: PropTypes.object,
  addToCont: PropTypes.func,
  handleViewTemplate: PropTypes.func,
  libraryProps: PropTypes.shape({
    ACTION_BUTTON: PropTypes.string,
    ACTION_BUTTON_BG: PropTypes.string,
    ACTION_BUTTON_BORDER: PropTypes.string,
    HEADER_TITLE: PropTypes.string,
    TEMPLATE_BACKGROUND: PropTypes.string,
    TEMPLATE_BORDER: PropTypes.string,
    TEMPLATE_DESCRIPTION: PropTypes.string,
    TEMPLATE_TITLE: PropTypes.string,
  }),
};

export default TemplateCard;
