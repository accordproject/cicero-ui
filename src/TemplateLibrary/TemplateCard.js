import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Image } from 'semantic-ui-react';

import TemplateActions from './TemplateActions';

const CardContainer = styled(Card)`
  margin: 10px 0 !important;
  border: ${props => props.tempborder || 'none'};
  border-radius: 6px !important;
  background-color: ${props => props.color || 'transparent'} !important;
  text-align: left;
  box-shadow: 0 1px 9px 0 rgba(0,0,0,0.1) !important;
`;

const Title = styled.div`
  display: inline;
  height: 16px;
  color: ${props => props.color || null};
  font-size: medium;
  font-weight: 600;
  line-height: 16px;
  margin-bottom: 5px;
`;

const TemplateLogo = styled(Image)`
  position: absolute !important;
  top: 13px;
  right: 16px;
  max-height: 23px;
`;

const Version = styled.span`
  margin-left: 10px;
  font-size: 0.75em;
  font-weight: 300;
`;

const DescriptionContainer = styled(Card.Description)`
  max-width: 400px;
  margin: auto;
  font-size: 0.79em;
  color: ${props => props.color || null} !important;
`;

/**
 * A Template Card component that will display the each template
 * and it's details.
 */
class TemplateCard extends React.Component {
  /**
     * Render this React component
     * @return {*} the react component
  */
  render() {
    const { libraryProps, template } = this.props;
    const displayName = template.displayName ? template.displayName : template.name;
    return (
        <CardContainer fluid
          key={template.uri}
          color={libraryProps.TEMPLATE_BACKGROUND}
          tempborder={libraryProps.TEMPLATE_BORDER}
        >
            <Card.Content>
              <TemplateLogo src={template.icon} />
              <Title color={libraryProps.TEMPLATE_TITLE}>
                {displayName}
                <Version>v {template.version}</Version>
              </Title>
              <DescriptionContainer color={libraryProps.TEMPLATE_DESCRIPTION}>
                {template.description}
              </DescriptionContainer>
            </Card.Content>
            <TemplateActions
              libraryProps={libraryProps}
              addToCont={this.props.addToCont}
              uriKey={template.uri}
              handleViewDetails={this.props.handleViewTemplate}
              className="templateAction"
            />
        </CardContainer>
    );
  }
}

/**
 * The property types for this component
 */
TemplateCard.propTypes = {
  template: PropTypes.object,
  addToCont: PropTypes.func,
  handleViewTemplate: PropTypes.func,
  libraryProps: PropTypes.shape({
    ACTION_BUTTON: PropTypes.string,
    ACTION_BUTTON_BG: PropTypes.string,
    ACTION_BUTTON_BORDER: PropTypes.string,
    HEADER_TITLE: PropTypes.string,
    TEMPLATE_BACKGROUND: PropTypes.string,
    TEMPLATE_BORDER: PropTypes.string,
    TEMPLATE_DESCRIPTION: PropTypes.string,
    TEMPLATE_TITLE: PropTypes.string,
  }),
};

export default TemplateCard;
