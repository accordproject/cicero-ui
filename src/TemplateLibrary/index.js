import _ from 'lodash';

/* React */
import React, { useState } from 'react';
import PropTypes from 'prop-types';

/* Styling */
import styled from 'styled-components';
import { Input } from 'semantic-ui-react';

/* Internal */
import TemplateCard from './TemplateCard';
import { ImportComponent, UploadComponent, NewClauseComponent } from './Buttons';

const TemplatesWrapper = styled.div`
  font-family: 'IBM Plex Sans', sans-serif;
  position: relative;
  max-width: 355px;
  height: inherit;

  grid-template-areas:  "header"
                        "functionTemps";
  grid-template-rows:    55px auto;
  grid-template-columns: 100%;
`;

const Header = styled.div`
  position: relative;
  font-family: 'IBM Plex Sans', sans-serif;
  font-weight: 800;
  font-size: 1em;
  margin-bottom: 10px;
  
  display: grid;
  grid-area: header;
  grid-template-columns: 1fr 1fr;
  grid-template-areas: "title imports";
`;

const HeaderTitle = styled.p`
  font-family: 'IBM Plex Sans', sans-serif;
  font-weight: 800;
  font-size: 1em;
  text-align: left;
  color: ${props => props.color || null};
`;

const HeaderImports = styled.div`
  display: flex;
  flex-direction: column;
`;

const Functionality = styled.div`
  font-family: 'IBM Plex Sans', sans-serif;
  max-width: 430px;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  grid-area: functionTemps;
  display: grid;
  max-height: 106px;
`;

const SearchInput = styled(Input)`
  margin: 5px 0 !important;
  width: 100% !important;
  max-height: 40px;
  border: 1px solid #939EBA;
  border-radius: 3px;
  box-shadow: inset 0 0 4px 0 #ABABAB;
`;

const TemplateCards = styled.div`
  margin: 0 !important;
  width: 100%;
  height: ${props => props.tempsHeight || 'calc(100% - 54px)'};
  display: inherit;
  overflow-y: scroll !important;
`;

/**
 * A Template Library component that will display the filtered list of templates
 * and provide drag-and-drop functionality.
 */

const TemplateLibraryComponent = (props) => {
  const [query, setQuery] = useState('');

  const onQueryChange = (e, input) => {
    const inputQuery = input.value.toLowerCase().trim();
    if (inputQuery !== query) {
      setQuery(inputQuery);
    }
  };

  const filterTemplates = (templates) => {
    let filteredTemplates = templates;
    if (query.length) {
      const regex = new RegExp(query, 'i');
      filteredTemplates = _.filter(filteredTemplates, t => (
        t.name.match(regex) || t.uri.match(regex)
      ));
    }
    return filteredTemplates;
  };

  /**
   * Render this React component
   * @return {*} the react component
   */
  const filtered = filterTemplates(props.templates);
  const libraryProps = props.libraryProps || Object.create(null);

  return (
      <TemplatesWrapper>
        <Header>
          <HeaderTitle color={libraryProps.HEADER_TITLE}>Clause Templates</HeaderTitle>
          <HeaderImports>
            {props.import
            && <ImportComponent importInput={props.import} />}
            {props.upload
            && <UploadComponent uploadInput={props.upload} />}
          </HeaderImports>
        </Header>
        <Functionality>
          <SearchInput className="icon" fluid icon="search" placeholder="Search..." onChange={onQueryChange} />
          {props.addTemp
          && <NewClauseComponent addTempInput={props.addTemp} />}
        </Functionality>
        <TemplateCards tempsHeight={libraryProps.TEMPLATES_HEIGHT} >
          {_.sortBy(filtered, ['name']).map(t => (
            <TemplateCard
              key={t.uri}
              addToCont={props.addToCont}
              template={t}
              handleViewTemplate={props.handleViewTemplate}
              className="templateCard"
              libraryProps={libraryProps}
            />
          ))}
        </TemplateCards>
      </TemplatesWrapper>
  );
};

TemplateLibraryComponent.propTypes = {
  libraryProps: PropTypes.shape({
    ACTION_BUTTON: PropTypes.string,
    ACTION_BUTTON_BG: PropTypes.string,
    ACTION_BUTTON_BORDER: PropTypes.string,
    HEADER_TITLE: PropTypes.string,
    TEMPLATE_BACKGROUND: PropTypes.string,
    TEMPLATE_BORDER: PropTypes.string,
    TEMPLATE_DESCRIPTION: PropTypes.string,
    TEMPLATE_TITLE: PropTypes.string,
    TEMPLATES_HEIGHT: PropTypes.string,
  }),
  upload: PropTypes.func,
  import: PropTypes.func,
  addTemp: PropTypes.func,
  addToCont: PropTypes.func,
  handleViewTemplate: PropTypes.func,
  templates: PropTypes.arrayOf(PropTypes.object),
};

export default TemplateLibraryComponent;
