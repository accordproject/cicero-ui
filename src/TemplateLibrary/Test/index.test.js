import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TemplateCard from '../Components/TemplateCard';
import TemplateLibrary from '../index';

const templateArray = [
  {
    description: 'This clause allows the receiver of goods to inspect them for a given time period after delivery.',
    name: 'acceptance-of-delivery',
    uri: 'ap://acceptance-of-delivery@0.11.0#311de48109cce10e6b2e33ef183ccce121886d0b76754d649d5054d1084f93cd',
    version: '0.11.0'
  },
  {
    description: 'a Simple Car Rental Contract in Turkish Language',
    name: 'car-rental-tr',
    uri: 'ap://car-rental-tr@0.8.0#1d6de2328745fe4bae726e0fb887fee6d3fbd00b3d717e30e38dc7bffd7222fa',
    version: '0.8.0'
  },
  {
    description: 'This clause is a copyright license agreement.',
    name: 'copyright-license',
    uri: 'ap://copyright-license@0.12.0#55d73ab5dacc642d8a7cd0bc95c7da5580bd031e00b955616837abd50be08b6b',
    version: '0.12.0'
  },
  {
    description: 'A sample demandforecast clause.',
    name: 'demandforecast',
    uri: 'ap://demandforecast@0.11.0#ac252106feb1f685f5ecceb12b967bb8210ca5283660b5bf72aeb5b397342f87',
    version: '0.11.0'
  },
  {
    description: 'Counts events from DocuSign connect with a given envelope status.',
    name: 'docusign-connect',
    uri: 'ap://docusign-connect@0.5.0#4b44811f4f3e4899c026c35f4180d6c555f9101e8fcca35a13ad09aa2e28b7f7',
    version: '0.5.0'
  },
  {
    description: 'This is a clause enforcing healthy eating habits in employees.',
    name: 'eat-apples',
    uri: 'ap://eat-apples@0.8.0#5d04f0bb8698e1e7565e496aa97beb055b1f3e7fb353d0b0447aeaac59729ed2',
    version: '0.8.0'
  },
];

const mockUpload = jest.fn();
const mockImport = jest.fn();
const addNewTemplate = jest.fn();
const mockAddToCont = jest.fn();

const propInput = {
  templates: templateArray,
  upload: mockUpload,
  import: mockImport,
  addTemp: addNewTemplate,
  addToCont: mockAddToCont,
};

const libraryProps = {
  ACTION_BUTTON: 'red',
  ACTION_BUTTON_BG: 'red',
  ACTION_BUTTON_BORDER: 'red',
  HEADER_TITLE: 'red',
  TEMPLATE_BACKGROUND: 'red',
  TEMPLATE_DESCRIPTION: 'red',
  TEMPLATE_TITLE: 'red',
};

describe('<TemplateLibrary />', () => {
  describe('on initialization', () => {
    it('renders page correctly', () => {
      const component = shallow(<TemplateLibrary {...propInput} libraryProps={libraryProps} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('renders conditional buttons', () => {
    it('with functions passed in', () => {
      const component = shallow(<TemplateLibrary libraryProps={libraryProps} {...propInput} />);
      expect(component.find('UploadComponent')).toHaveLength(1);
      expect(component.find('ImportComponent')).toHaveLength(1);
      expect(component.find('NewClauseComponent')).toHaveLength(1);
    });

    it('without functions passed in', () => {
      const component = shallow(<TemplateLibrary libraryProps={libraryProps} />);
      expect(component.find('UploadComponent')).toHaveLength(0);
      expect(component.find('ImportComponent')).toHaveLength(0);
      expect(component.find('NewClauseComponent')).toHaveLength(0);
    });
  });

  describe('runs functions passed into it', () => {
    it('add To Contract function runs', () => {
      const component = shallow(
        <TemplateCard
          key={templateArray[0].uri}
          template={templateArray[0]}
          addToCont={mockAddToCont}
          libraryProps={libraryProps}
        />
      );
      expect(component.find('.templateAction').prop('addToCont')).toEqual(mockAddToCont);
    });
  });
});
