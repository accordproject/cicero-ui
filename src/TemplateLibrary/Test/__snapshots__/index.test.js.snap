// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateLibrary /> on initialization renders page correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.p
      color="red"
    >
      Clause Templates
    </styled.p>
    <styled.div>
      <ImportComponent
        importInput={[MockFunction]}
      />
      <UploadComponent
        uploadInput={[MockFunction]}
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <Styled(Input)
      className="icon"
      fluid={true}
      icon="search"
      onChange={[Function]}
      placeholder="Search..."
    />
    <NewClauseComponent
      addTempInput={[MockFunction]}
    />
  </styled.div>
  <styled.div>
    <TemplateCard
      addToCont={[MockFunction]}
      className="templateCard"
      key="ap://acceptance-of-delivery@0.11.0#311de48109cce10e6b2e33ef183ccce121886d0b76754d649d5054d1084f93cd"
      libraryProps={
        Object {
          "ACTION_BUTTON": "red",
          "ACTION_BUTTON_BG": "red",
          "ACTION_BUTTON_BORDER": "red",
          "HEADER_TITLE": "red",
          "TEMPLATE_BACKGROUND": "red",
          "TEMPLATE_DESCRIPTION": "red",
          "TEMPLATE_TITLE": "red",
        }
      }
      template={
        Object {
          "description": "This clause allows the receiver of goods to inspect them for a given time period after delivery.",
          "name": "acceptance-of-delivery",
          "uri": "ap://acceptance-of-delivery@0.11.0#311de48109cce10e6b2e33ef183ccce121886d0b76754d649d5054d1084f93cd",
          "version": "0.11.0",
        }
      }
    />
    <TemplateCard
      addToCont={[MockFunction]}
      className="templateCard"
      key="ap://car-rental-tr@0.8.0#1d6de2328745fe4bae726e0fb887fee6d3fbd00b3d717e30e38dc7bffd7222fa"
      libraryProps={
        Object {
          "ACTION_BUTTON": "red",
          "ACTION_BUTTON_BG": "red",
          "ACTION_BUTTON_BORDER": "red",
          "HEADER_TITLE": "red",
          "TEMPLATE_BACKGROUND": "red",
          "TEMPLATE_DESCRIPTION": "red",
          "TEMPLATE_TITLE": "red",
        }
      }
      template={
        Object {
          "description": "a Simple Car Rental Contract in Turkish Language",
          "name": "car-rental-tr",
          "uri": "ap://car-rental-tr@0.8.0#1d6de2328745fe4bae726e0fb887fee6d3fbd00b3d717e30e38dc7bffd7222fa",
          "version": "0.8.0",
        }
      }
    />
    <TemplateCard
      addToCont={[MockFunction]}
      className="templateCard"
      key="ap://copyright-license@0.12.0#55d73ab5dacc642d8a7cd0bc95c7da5580bd031e00b955616837abd50be08b6b"
      libraryProps={
        Object {
          "ACTION_BUTTON": "red",
          "ACTION_BUTTON_BG": "red",
          "ACTION_BUTTON_BORDER": "red",
          "HEADER_TITLE": "red",
          "TEMPLATE_BACKGROUND": "red",
          "TEMPLATE_DESCRIPTION": "red",
          "TEMPLATE_TITLE": "red",
        }
      }
      template={
        Object {
          "description": "This clause is a copyright license agreement.",
          "name": "copyright-license",
          "uri": "ap://copyright-license@0.12.0#55d73ab5dacc642d8a7cd0bc95c7da5580bd031e00b955616837abd50be08b6b",
          "version": "0.12.0",
        }
      }
    />
    <TemplateCard
      addToCont={[MockFunction]}
      className="templateCard"
      key="ap://demandforecast@0.11.0#ac252106feb1f685f5ecceb12b967bb8210ca5283660b5bf72aeb5b397342f87"
      libraryProps={
        Object {
          "ACTION_BUTTON": "red",
          "ACTION_BUTTON_BG": "red",
          "ACTION_BUTTON_BORDER": "red",
          "HEADER_TITLE": "red",
          "TEMPLATE_BACKGROUND": "red",
          "TEMPLATE_DESCRIPTION": "red",
          "TEMPLATE_TITLE": "red",
        }
      }
      template={
        Object {
          "description": "A sample demandforecast clause.",
          "name": "demandforecast",
          "uri": "ap://demandforecast@0.11.0#ac252106feb1f685f5ecceb12b967bb8210ca5283660b5bf72aeb5b397342f87",
          "version": "0.11.0",
        }
      }
    />
    <TemplateCard
      addToCont={[MockFunction]}
      className="templateCard"
      key="ap://docusign-connect@0.5.0#4b44811f4f3e4899c026c35f4180d6c555f9101e8fcca35a13ad09aa2e28b7f7"
      libraryProps={
        Object {
          "ACTION_BUTTON": "red",
          "ACTION_BUTTON_BG": "red",
          "ACTION_BUTTON_BORDER": "red",
          "HEADER_TITLE": "red",
          "TEMPLATE_BACKGROUND": "red",
          "TEMPLATE_DESCRIPTION": "red",
          "TEMPLATE_TITLE": "red",
        }
      }
      template={
        Object {
          "description": "Counts events from DocuSign connect with a given envelope status.",
          "name": "docusign-connect",
          "uri": "ap://docusign-connect@0.5.0#4b44811f4f3e4899c026c35f4180d6c555f9101e8fcca35a13ad09aa2e28b7f7",
          "version": "0.5.0",
        }
      }
    />
    <TemplateCard
      addToCont={[MockFunction]}
      className="templateCard"
      key="ap://eat-apples@0.8.0#5d04f0bb8698e1e7565e496aa97beb055b1f3e7fb353d0b0447aeaac59729ed2"
      libraryProps={
        Object {
          "ACTION_BUTTON": "red",
          "ACTION_BUTTON_BG": "red",
          "ACTION_BUTTON_BORDER": "red",
          "HEADER_TITLE": "red",
          "TEMPLATE_BACKGROUND": "red",
          "TEMPLATE_DESCRIPTION": "red",
          "TEMPLATE_TITLE": "red",
        }
      }
      template={
        Object {
          "description": "This is a clause enforcing healthy eating habits in employees.",
          "name": "eat-apples",
          "uri": "ap://eat-apples@0.8.0#5d04f0bb8698e1e7565e496aa97beb055b1f3e7fb353d0b0447aeaac59729ed2",
          "version": "0.8.0",
        }
      }
    />
  </styled.div>
</styled.div>
`;
